iroh-net/src/magic_endpoint.rs

```diff
@@ -380,6 +352,11 @@ impl MagicEndpoint {
         self.msock.local_endpoints().await
     }
 
+    /// Returns the next non empty version of `local_endpoints`.
+    pub async fn ensure_local_endpoints(&self) -> Result<Vec<config::Endpoint>> {
```

Comment:
why not build this into the `local_endpoints()` call instead of adding a new API?  The first time it is called it could ensure that it waits - it already is async.

The address-related API of MagicEndpoint is already large, and this addition does not make it clearer.  This is supposed to be the high-level-easy-to-use API...

Reply:
yes, agreed, fixed